httpdocs/ajax/animation/create.php

```diff
@@ -140,14 +144,24 @@ function parse_mf_string($line, $parameter) {
     error('Cannot move scene file.');
   if ($thumbnailAvailable && !move_uploaded_file($_FILES['thumbnail-file']['tmp_name'], "$folder/thumbnail.jpg"))
     error('Cannot move thumbnail file.');
-  if ($total > 0) {
+  if ($total_textures > 0) {
     mkdir("$folder/textures");
-    for($i = 0; $i < $total; $i++) {
+    for($i = 0; $i < $total_textures; $i++) {
       $target = basename($_FILES['textures']['name'][$i]);
       if ($target == '')
         continue;
       if (!move_uploaded_file($_FILES['textures']['tmp_name'][$i], "$folder/textures/$target"))
-        error("Cannot move $total $target");
+        error("Cannot move $total_textures $target");
+    }
+  }
+  if ($total_meshes > 0) {
+    mkdir("$folder/meshes");
+    for($i = 0; $i < $total_meshes; $i++) {
+      $target = basename($_FILES['meshes']['name'][$i]);
+      if ($target == '')
+        continue;
+      if (!move_uploaded_file($_FILES['meshes']['tmp_name'][$i], "$folder/meshes/$target"))
+        error("Cannot move $total_meshes $target");
```

Comment:
Instead of duplicating the code, it would be better to create a new function and call it twice.